scripts/Packages/Package_BGA/bga.py

```diff
@@ -153,6 +153,19 @@ def crtYdRound(x):
 
     # Pads
     balls = layoutX * layoutY
+    
+    # use IPC-derived pad size if possible, then fall back to user-defined pads
+    if (("ball_type" in fpParams) and ("ball_diameter" in fpParams)):
+        padSize = configuration['collapsible'][fpParams["ball_diameter"]]["min"]
```

Comment:
why is `collapsible` hard-coded

Reply:
What happens if that ball_diameter is not found in the `.yaml` file?
I think some checking if the required values are present is needed here.

Reply:
Whoops! Fixed the ball type being fixed and not read from the YAML file.

I made a separate function to handle argument hygiene and then pass onto the footprint generation. Much like the gullwing generator script. This would also allow an extension of the script to generate footprints with multiple IPC densities or something (this isn't done on any of the Package* scripts but it applies to all of them).